xir/parser.py

```diff
@@ -192,10 +193,17 @@ def gate_def(self, args):
                 if has_declared_wires:
                     stmts = args[i:]
                     break
-
                 stmts.append(arg)
-                for w in arg.wires:
-                    wires += (w,) if w not in wires else ()
+
+                int_wires = [w for w in arg.wires if isinstance(w, int)]
+                wires += tuple(arg.wires)
```

Comment:
Do we actually need this line given that we're assigning `wires` to a function of `max_wire` below?
```suggestion
```

Reply:
This wasn't actually super-clear before, but I agree it could be removed now. Before my last commit this was needed to still get a reasonable program before running the validation (I'm not sure if it makes sense to think that way though). If string wires were used inside a definition but _not_ declared, this would declare them implicitly. I changed this so that only integer wires are implicitly declared.